### PR TITLE
Refactoring: refactoring and move resource file absLvl.

### LINKED ISSUE #72: Refactoring of the abstract level
do after #71

### PR DESCRIPTION
**Pull request by IgorekLoschinin**
fix #72 Refactoring abstract level.

### CODE REVIEW

**Comment by EndrII** [comment]
Fix windows build before my review

File: src/Client/CMakeLists.txt [hunk 1]
```diff
@@ -27,6 +27,7 @@ else()
 endif()
 
 target_link_libraries(${CURRENT_PROJECT} PUBLIC ${PROJECT_NAME}Core TestLvl)
+target_link_libraries(${CURRENT_PROJECT} PUBLIC ${PROJECT_NAME}Core AbstractLvl)
```
Comment (EndrII) [inline src/Client/CMakeLists.txt:30]:
```suggestion
target_link_libraries(${CURRENT_PROJECT} PUBLIC ${PROJECT_NAME}Core TestLvl AbstractLvl)
```

use on line for include libraries

File: src/CrawlAbstractLvl/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Copyright (C) 2020-2021 QuasarApp.
+# Distributed under the lgplv3 software license, see the accompanying
+# Everyone is permitted to copy and distribute verbatim copies
+# of this license document, but changing it is not allowed.
+#
+
+cmake_minimum_required(VERSION 3.14)
+set(CURRENT_PROJECT "AbstractLvl")
+
+add_definitions(-DSnakeProject_LIBRARY)
```
Comment (EndrII) [inline src/CrawlAbstractLvl/CMakeLists.txt:11]: you must change the default library define to CRAWL_ABSTRACT_LEVEL_LIBRARY define
and do not forget change in in your global.h file

File: src/CrawlAbstractLvl/private/abslvlcontrol.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#ifndef TESTCONTROL_H
+#define TESTCONTROL_H
+#include "Crawl/defaultcontrol.h"
+
+/**
+ * @brief The TestControl class This controll support custom camera-ratation functions.
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/abslvlcontrol.h:13]: The description of the class douse not correspond of reality.

File: src/CrawlAbstractLvl/private/abslvlcontrol.h [hunk 4]
```diff
@@ -0,0 +1,24 @@
+//#
+//# Copyright (C) 2021-2021 QuasarApp.
+//# Distributed under the GPLv3 software license, see the accompanying
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#ifndef TESTCONTROL_H
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/abslvlcontrol.h:8]: you use duplicate macros with the test lvl library.

File: src/CrawlAbstractLvl/private/abslvlworld.cpp [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#include <snake.h>
+#include "Crawl/iworlditem.h"
+
+
+AbsLvlWorld::AbsLvlWorld() {
+
+}
+
+IPlayer *AbsLvlWorld::initPlayer() const {
+    return new Snake();
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/abslvlworld.cpp:22]: you should create a own Snake class item. For more information see the test lvl.

File: src/CrawlAbstractLvl/private/abslvlworld.h [hunk 6]
```diff
@@ -0,0 +1,32 @@
+//#
+//# Copyright (C) 2021-2021 QuasarApp.
+//# Distributed under the GPLv3 software license, see the accompanying
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#ifndef WORLD_H
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/abslvlworld.h:8]: please change this define name to the name of the class .
example : 
ABSLEVELWORLD_H

File: src/CrawlAbstractLvl/private/obstaclered.cpp [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+//# Distributed under the GPLv3 software license, see the accompanying
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#include "obstaclered.h"
+
+ObstacleRed::ObstacleRed() : IWorldItem("ObstacleRed") {
+    setMash("qrc:/mesh/meshes/ObstacleRed.mesh");
+    setSize({10,10,10});
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/obstaclered.cpp:12]: the 10 size it is very gib size. So use the 1 size.

File: src/CrawlAbstractLvl/private/snake.h [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#ifndef SNAKE_H
+#define SNAKE_H
+
+#include "Crawl/iplayer.h"
+
+class Snake : public IPlayer {
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/snake.h:13]: The snake object must be  inherit of the Snake class in core module.

File: src/CrawlAbstractLvl/private/snake.h [hunk 9]
```diff
@@ -0,0 +1,30 @@
+//#
+//# Copyright (C) 2021-2021 QuasarApp.
+//# Distributed under the GPLv3 software license, see the accompanying
+//# Everyone is permitted to copy and distribute verbatim copies
+//# of this license document, but changing it is not allowed.
+//#
+
+#ifndef SNAKE_H
```
Comment (EndrII) [inline src/CrawlAbstractLvl/private/snake.h:8]: This class will be conflict with Snake class of the core module. so please rename your snake class.

**Review by EndrII — CHANGES_REQUESTED** [review]

**Comment by EndrII** [comment]
@Oleg-designer  please build this branch and make manual tests of this level. 
After test add new issues  and attach them to this pull request.

**Comment by EndrII** [comment]
@IgorekLoschinin This look good for me but i think that this level still not works correctly.

**Comment by Oleg-designer** [comment]
@IgorekLoschinin  When i try to start game they crush with this text in console "Segmentation fault (core dumped)" need to fix this.

**Comment by EndrII** [comment]
please import last changes from main into this branch (using merge strategy)

**Review by EndrII — APPROVED** [review]
Will be merget

**Pull request merged**